rosetta-lang/src/main/java/com/regnosys/rosetta/validation/RosettaSimpleValidator.xtend

```diff
@@ -137,8 +158,49 @@ class RosettaSimpleValidator extends AbstractDeclarativeRosettaValidator {
 	@Inject extension RBuiltinTypeService
 	@Inject extension TypeSystem
 	@Inject extension RosettaGrammarAccess
+	@Inject extension TypeValidationUtil
 	@Inject extension RObjectFactory objectFactory
 
+
+	@Check
+	def void switchStatementMustProvideCaseForAllEnumValues(SwitchOperation op) {
+		val argumentType = op.argument.RType
+		if (op.^default === null && argumentType instanceof REnumType) {
+			val enumConditions = op.values.map[it.enumCondition].toSet
+
+			val enumeration = (argumentType as REnumType).EObject
+			for (enumValue : enumeration.enumValues) {
+				if (!enumConditions.contains(enumValue)) {
+					error("Not all possible values of enumeration provided to switch, either provide all or use default", enumValue, null)
+				}
+			}
+		}
+
+	}
+
+	@Check
+ 	def void switchArgumentTypeMatchesCaseStatmentTypes(SwitchOperation op) {
+ 		val argumentRType = op.argument.RType
+ 		for (CaseStatement caseStatement : op.values) {
+ 			if (caseStatement.literalCondition !== null) {
+ 				val conditionType = caseStatement.literalCondition.RType
+	 			if (!conditionType.isSubtypeOf(argumentRType)) {
+ 					error('''Mismatched condition type: «argumentRType.notASubtypeMessage(conditionType)»''', caseStatement.literalCondition ?: caseStatement.enumCondition , null)
+ 				}
+ 			} 
+
+ 		}
+ 	}
+
+ 	@Check
+ 	def void switchArgumentsAreCorrectTypes(SwitchOperation op) {
+ 		val inputType = op.argument.RType
+ 		if (inputType instanceof RDataType) {
+ 			val message = "Invalid switch argument type, supported argument types are basic types and enumerations"
+ 			error(message, op.argument, null)
+ 		}
```

Comment:
What if the input to a switch operation is of multi cardinality?

Reply:
Also: I guess we shouldn't allow record types. Perhaps check that it's not a basic type and not an enum type? (Stripped from aliases)

Also: it is often useful for the user to display the type in the message.